spec/datadog/core/process_discovery_spec.rb

```diff
@@ -26,14 +32,14 @@
         end
 
         it 'stores metadata successfully' do
-          native_fd = described_class.get_and_store_metadata(settings, Datadog::Core::Logger.new(StringIO.new))
+          described_class.get_and_store_metadata(settings, Datadog::Core::Logger.new(StringIO.new))
+          native_fd = described_class.instance_variable_get(:@file_descriptor)
 
           # Extract content from created memfd
           fd = described_class._native_to_rb_int(native_fd)
-          buffer = IO.new(fd)
+          buffer = IO.new(fd, autoclose: false)
```

Comment:
Since the `autoclose: false` is both:
* **really important to not make our test-suite flaky**

as well as

* **extremely non-obvious**

it's one more thing that's useful to leave behind a comment saying why we're doing this, e.g. `# This MUST be autoclose: false because our code handles closing separately and if Ruby also tries to close the wrong file descriptor can be closed, potentially creating flaky specs and other bad things"